perf/src/sigverify.rs

```diff
@@ -1350,40 +911,45 @@ mod tests {
         {
             let mut tx = test_tx();
             tx.message.instructions[0].data = vec![1, 2, 3];
-            let mut packet = Packet::from_data(None, tx).unwrap();
-            let packet_offsets = do_get_packet_offsets(&packet, 0).unwrap();
-            check_for_simple_vote_transaction(&mut packet, &packet_offsets, 0).ok();
-            assert!(!packet.meta().is_simple_vote_tx());
+            let packet = Packet::from_data(None, tx).unwrap();
+            let packet_bytes = packet.data(..).unwrap();
+            let transaction_meta = TransactionMeta::try_new(packet_bytes).unwrap();
+            assert!(!is_simple_vote_tx(packet_bytes, &transaction_meta));
         }
 
         // single legacy vote tx is
         {
             let mut tx = new_test_vote_tx(&mut rng);
             tx.message.instructions[0].data = vec![1, 2, 3];
-            let mut packet = Packet::from_data(None, tx).unwrap();
-            let packet_offsets = do_get_packet_offsets(&packet, 0).unwrap();
-            check_for_simple_vote_transaction(&mut packet, &packet_offsets, 0).ok();
-            assert!(packet.meta().is_simple_vote_tx());
+            let packet = Packet::from_data(None, tx).unwrap();
+            let packet_bytes = packet.data(..).unwrap();
+            let transaction_meta = TransactionMeta::try_new(packet_bytes).unwrap();
+            assert!(is_simple_vote_tx(packet_bytes, &transaction_meta));
         }
 
         // single versioned vote tx is not
         {
             let mut tx = new_test_vote_tx(&mut rng);
             tx.message.instructions[0].data = vec![1, 2, 3];
+
+            let msg_start = {
+                1 + // num signatures
+                tx.signatures.len() * core::mem::size_of::<Signature>() // signatures
+            };
+
             let mut packet = Packet::from_data(None, tx).unwrap();
 
-            // set messager version to v0
-            let mut packet_offsets = do_get_packet_offsets(&packet, 0).unwrap();
-            let msg_start = packet_offsets.msg_start as usize;
+            // set message version to v0
             let msg_bytes = packet.data(msg_start..).unwrap().to_vec();
             packet.buffer_mut()[msg_start] = MESSAGE_VERSION_PREFIX;
-            packet.meta_mut().size += 1;
-            let msg_end = packet.meta().size;
+            let msg_end = packet.meta().size + 1;
             packet.buffer_mut()[msg_start + 1..msg_end].copy_from_slice(&msg_bytes);
+            packet.meta_mut().size += 2; // 1 byte for version, 1 byte for num ATLs
```

Comment:
made code comments where there were changes.
These tests were previously **incorrectly** changing the serialization.
Previously sigverify would not check serialization of ATLs, but now it does. That was leading the below `unwrap` of `TransactionMeta` failing, even though it was correctly identifying the packet was malformed.

These test's modification of the buffers neglected that when we change to `v0` we must add 2 bytes, not 1.
- 1 byte is for the message prefix (v0).
- 1 byte is for the number of ATL (in all these cases 0)